packages/amplify-prompts/src/progressbars/multibar.ts

```diff
@@ -0,0 +1,240 @@
+    private timer!: ReturnType<typeof setTimeout>;
+    private prefixText : string;
+    private updated: boolean;
+    private lastDrawnStrings : string[];
+
+    constructor(options : BarOptions) {
+      this.terminal = new Terminal();
+      this.options = options;
+      this.bars = [];
+
+      if (this.options.hideCursor === true) {
+        this.terminal.cursor(false);
+      }
+
+      this.lastDrawnTime = Date.now();
+      this.isActive = false;
+      this.count = 0;
+
+      this.frameCount = 0;
+
+      // simulate spinner
+      this.frames = ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'];
+      this.prefixText = options.prefixText;
+      this.refreshRate = 200;
+
+      // Tracks when an update into one of the the bars happens
+      this.updated = false;
+      this.lastDrawnStrings = [];
+    }
+
+    /**
+     * Checks if the environment is tty
+     */
+    isTTY() : boolean {
+      return this.terminal.isTTY();
+    }
+
+    /**
+     * Writes lines into the re writable block
+     */
+    writeLines(prefixText: string) : void {
+      let barStrings : string[] = [];
+      let stringsToRender : string[] = [];
+      if (prefixText.length) {
+        stringsToRender.push(prefixText);
+      }
+      // Only call on the render strings for the individual bar if an update happened.
+      if (this.updated) {
+        barStrings = this.bars.reduce((prev, _current) => prev.concat(_current.bar.getRenderStrings()), barStrings);
+        stringsToRender = stringsToRender.concat(barStrings);
+        this.lastDrawnStrings = barStrings;
+      } else {
+        // Use the cached last drawn strings if no update
+        stringsToRender = stringsToRender.concat(this.lastDrawnStrings);
+      }
+      this.terminal.writeLines(stringsToRender);
+    }
+
+    /**
+     * Render function which is called repeatedly
+     */
+    render() : void {
+      let initLine = '';
+      if (this.timer) {
+        clearTimeout(this.timer);
+      }
+      // Init line is prefix text plus spinner
+      if (this.prefixText.length) {
+        initLine = `${this.prefixText} ${this.frames[this.frameCount]}`;
+      }
+      this.writeLines(initLine);
+
+      this.frameCount = ++this.frameCount % this.frames.length;
+      this.lastDrawnTime = Date.now();
+
+      // Enter a newLine in nonTTY mode
+      if (!this.isTTY()) {
+        this.terminal.newLine();
+      }
+      // Reset updated flag
+      this.updated = false;
+
+      this.timer = setTimeout(this.render.bind(this), this.refreshRate);
+    }
+
+    /**
+     * Returns bar indexed by name
+     */
+    getBar(name: string) : { name: string, bar: Bar } | undefined {
+      return this.bars.find(obj => obj.name === name);
+    }
+
+    /**
+     * Updates a progress bar by adding/updating item or increments the bar
+     */
+    updateBar(name: string, updateObj: { name: string, payload: ItemPayload }) : void {
+      const barDetails = this.getBar(name);
+      if (barDetails) {
```

Comment:
prefer to flip these conditions and early return. eg:
```
if (!barDetails) {
  return;
}
// do other stuff
```

Applies to other places in this class as well